autoload/sy.vim

```diff
@@ -106,7 +106,16 @@ endfunction
 
 " #buffer_is_active {{{1
 function! sy#buffer_is_active()
-  return !empty(getbufvar(bufnr(''), 'sy'))
+  return sy#buffer_is_active_bnr('')
+endfunction
+
+" #buffer_is_active_bnr {{{1
+function! sy#buffer_is_active_bnr(bnr)
+  let l:bufnr = bufnr('')
+  if type(a:bnr) == type(123) && a:bnr > 0
+    let l:bufnr = a:bnr
+  endif
```

Comment:
I'd rather throw error if argument is incorrect. This can help to find potential bugs instead of hiding them. Basically it's better to not have default value here.

Also `123` looks like a magic number, I think `type(0)` is cleaner.

Reply:
pls look over others funcs of this plugins, it even no any arguments check...
but it should be ok, basically those called was on purposed and private...
and again, just for compatible and protective modification, or actually new version viml can use 'argument default', no needed such ugly.